### PR TITLE
Code snippet displayer

### PR DESCRIPTION
**Pull request by vittoriaThinkst**
## Proposed changes

Add Code Snippet displayer ( single line, multiline )
Add Refresh Button
Add Unit tests for both 

Screenshot 2024-03-27 at 14 32 38

https://github.com/thinkst/canarytokens/assets/126554007/d1ea8b97-cb6e-4f7f-a603-977824be5b45

### CODE REVIEW

File: frontend_vue/src/components/base/BaseCodeSnippet.vue [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+          @refresh-token="handleRefreshToken"
+        />
+        <BaseCopyButton :content="code" />
+      </div>
+      <VCodeBlock
+        :id="label"
+        :code="code"
+        highlightjs
+        :lang="lang"
+        theme="a11y-light"
```
Comment (mclmax) [inline frontend_vue/src/components/base/BaseCodeSnippet.vue:23]: do we like this theme? it feels a little christmassy. what about something more githubesque?

File: frontend_vue/src/components/base/BaseCodeSnippet.vue [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      class="inline-block mb-8 text-grey-500"
+      :for="label"
+      >{{ label }}</label
+    >
+
+    <div class="relative border rounded-lg border-grey-100">
+      <div class="absolute top-[.8rem] right-[1rem] z-10 flex gap-8">
+        <BaseRefreshButton
+          v-if="hasRefresh"
+          @refresh-token="handleRefreshToken"
```
Comment (mclmax) [inline frontend_vue/src/components/base/BaseCodeSnippet.vue:14]: what is this for exactly?

File: frontend_vue/src/components/base/BaseRefreshButton.vue [hunk 3]
```diff
@@ -0,0 +1,88 @@
+<template>
```
Comment (mclmax) [inline frontend_vue/src/components/base/BaseRefreshButton.vue:1]: what is this component used for?

File: frontend_vue/src/components/base/BaseCodeSnippet.vue (hunk 1 above)
Comment (vittoriaThinkst) [reply frontend_vue/src/components/base/BaseCodeSnippet.vue:23]: christmassy 😂 no yeah for js it's true, it also changes depending on the language
( this for instance is the snippet for php 
Screenshot 2024-03-27 at 15 04 23

But we can use another one, I'm not opinionated about it
https://webdevnerdstuff.github.io/vue-code-block/#props

I would just keep one with a light background

File: frontend_vue/src/components/base/BaseCodeSnippet.vue (hunk 2 above)
Comment (vittoriaThinkst) [reply frontend_vue/src/components/base/BaseCodeSnippet.vue:14]: We have this functionality in some places
Screenshot 2024-03-27 at 15 08 02

( it looks like it's broken on prod 👀 is it just me? )

File: frontend_vue/src/components/base/BaseRefreshButton.vue (hunk 3 above)
Comment (vittoriaThinkst) [reply frontend_vue/src/components/base/BaseRefreshButton.vue:1]: see above

File: frontend_vue/src/components/base/BaseCodeSnippet.vue (hunk 1 above)
Comment (mclmax) [reply frontend_vue/src/components/base/BaseCodeSnippet.vue:23]: can we try the github theme?

File: frontend_vue/src/components/base/BaseCodeSnippet.vue (hunk 2 above)
Comment (mclmax) [reply frontend_vue/src/components/base/BaseCodeSnippet.vue:14]: oh i see... yes it looks broken eish. ok we can fix it in this version heh

File: frontend_vue/src/components/base/BaseCodeSnippet.vue (hunk 1 above)
Comment (vittoriaThinkst) [reply frontend_vue/src/components/base/BaseCodeSnippet.vue:23]: I think it looks fine 

JS:
Screenshot 2024-03-27 at 15 19 50

**Review by mclmax — APPROVED** [review]

**Pull request merged**